rotkehlchen/chain/ethereum/modules/l2/zksync.py

```diff
@@ -0,0 +1,589 @@
+
+            log.error(
+                f'Could not deserialize zksync lite transaction {entry} due to {error}',
+            )
+        else:
+            return ZKSyncLiteTransaction(
+                tx_hash=tx_hash,
+                tx_type=tx_type,
+                timestamp=timestamp,
+                block_number=block_number,
+                from_address=from_address,
+                to_address=to_address,
+                asset=asset,
+                amount=amount,
+                fee=Fee(asset_normalized_value(fee_raw, asset)) if fee_raw else None,
+            )
+
+        return None
+
+    def _query_zksync_api_transactions(
+            self,
+            address: ChecksumEvmAddress,
+            from_hash: str,
+            direction: Literal['older', 'newer'],
+    ) -> Iterator[list[ZKSyncLiteTransaction]]:
+        transactions = []
+        last_tx_hash = ''
+        while True:
+            options = {'from': from_hash, 'limit': ZKSYNCLITE_MAX_LIMIT, 'direction': direction}
+            response = self._query_api(
+                url=f'accounts/{address}/transactions',
+                options=options,
+            )
+            result = response.get('list', None)
+            if result is None:
+                msg = 'Unexpected zksync transactions response. Could not find result in the response.'  # noqa: E501
+                log.error(f'{msg} Response: {response}')
+                raise RemoteError(msg)
+
+            for idx, entry in enumerate(result):
+                if idx == 0 and entry['txHash'] == last_tx_hash:
+                    continue  # at pagination first tx is last query's last
+
+                tx = self._decode_zksync_transaction(entry)
+                if tx:
+                    transactions.append(tx)
+                    last_tx_hash = entry['txHash']
+
+            yield transactions
+            if len(result) < ZKSYNCLITE_MAX_LIMIT:
+                break  # no need to paginate further
+
+            from_hash = result[-1]['txHash']
+            transactions = []
+
+    def _add_zksynctxs_db(
+            self,
+            write_cursor: 'DBCursor',
+            transactions: Iterable[ZKSyncLiteTransaction],
+    ) -> None:
+        tuples = [x.serialize_for_db() for x in transactions]
+        query = """INSERT INTO zksynclite_transactions(tx_hash, type, timestamp, block_number,
+        from_address, to_address, token_identifier, amount, fee)
+        VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)"""
+        self.database.write_tuples(
+            write_cursor=write_cursor,
+            tuple_type='zksynclite_transaction',
+            query=query,
+            tuples=tuples,
+        )
+
+    def _get_zksynctxs_db(
+            self,
+            address: ChecksumEvmAddress,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[ZKSyncLiteTransaction]:
+        cursor = self.database.conn.cursor()
+        result = cursor.execute(
```

Comment:
why not use the with statement? You need to close the cursor manually otherwise (?)

Reply:
Because I guess when I wrote it there was no with statement ;)